### PR TITLE
Fix MG PLC algos intermittent hang

### LINKED ISSUE #2568: [BUG] Failure occurs when running mg PLC algos numerous times
**Describe the bug**
An intermittent failure occurs when running mg PLC algos like `core-number` and `PageRank` several times with the same `PLC Graph`.

**Steps/Code to reproduce bug**
run mg `core-number` or `PageRank` more than once with the same `PLC Graph`

### PR DESCRIPTION
**Pull request by jnke2016**
Dask doesn't always release some of the inactive futures fast enough. This can be problematic when running the same `algo` several times with the same `PLC graph` because those futures can be cache in the next iteration causing a hang if some get released midway.

This PR manually delete inactive futures.
closes #2568

### CODE REVIEW

**Review by rlratzel — APPROVED** [review]

File: python/cugraph/cugraph/dask/cores/core_number.py [hunk 1]
```diff
@@ -126,7 +127,16 @@ def core_number(input_graph,
 
     wait(cudf_result)
 
-    ddf = dask_cudf.from_delayed(cudf_result)
+    # FIXME: Dask doesn't always release it fast enough.
+    # For instance if the algo is run several times with
+    # the same PLC graph, the current iteration might try to cache
+    # the past iteration's futures and this can cause a hang if some
+    # of those futures get released midway
+    del result
```
Comment (VibhuJawa) [inline python/cugraph/cugraph/dask/cores/core_number.py:135]: I think removing `result` is still a good practice as when `cudf_result`  is completed  we will still have `result` in memory so from a memory relief standpoint it makes sense.

File: python/cugraph/cugraph/dask/cores/core_number.py [hunk 2]
```diff
@@ -126,7 +127,16 @@ def core_number(input_graph,
...
+
+    ddf = dask_cudf.from_delayed(cudf_result).persist()
+    wait(ddf)
+
```
Comment (VibhuJawa) [inline python/cugraph/cugraph/dask/cores/core_number.py:139]: I suggest we also remove `cudf_result` here to save on memory . We have a renumbered call after this so `result` does not go out of scope immediately.

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
I am requesting to delete `cudf_result`, looks good otherwise.

**Comment by jnke2016** [comment]
rerun tests

**Comment by jnke2016** [comment]
rerun tests

**Comment by jnke2016** [comment]
rerun tests

**Comment by jnke2016** [comment]
rerun tests

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**